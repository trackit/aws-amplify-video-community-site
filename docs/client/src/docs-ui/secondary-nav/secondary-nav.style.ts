import {css} from "emotion";
import {MQLaptop, MQTablet, MQMobile} from "../../amplify-ui/styles/media";

export const secondaryNavStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  flex: 1;
  align-items: center;

  ${MQLaptop} {
    flex-direction: row;
    padding-left: 1.5rem;
  }

  * {
    white-space: nowrap;
  }

  > div {
    display: flex;
    flex-direction: row;
    align-items: center;
    box-sizing: border-box;

    &:first-child {
      width: 100%;
      overflow-x: auto;

      > div {
        display: flex;
        flex-direction: row;
        padding: 0 1rem;
      }
    }

    docs-internal-link,
    amplify-external-link {
      margin-left: 1.75rem;

      &:first-child {
        margin-left: 0;
      }

      a {
        display: block;
        padding: 0.875rem 0 0.875rem;
        color: var(--font-color);
      }
    }
  }
`;

export const spanStyle = css`
  color: var(--font-color-secondary);
`;

export const linkActiveStyle = css`
  > a {
    font-weight: bold;

    ${MQLaptop} {
      border-bottom: 0.125rem solid var(--font-color);
    }

    > span {
      color: var(--font-color);
    }
  }
`;

export const hostStyle = css`
  z-index: 1;
  display: block;
  position: sticky;
  top: 0;
  background-color: var(--bg-color-secondary);

  ${MQTablet} {
    padding: 0;
  }
`;

export const ghostItemStyle = css`
  display: block;
  width: 1rem;
  height: 100%;
`;

export const shadowStyle = css`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  width: 2rem;
  background-image: linear-gradient(
    to right,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0.1875)
  );

  ${MQLaptop} {
    display: none;
  }
`;
