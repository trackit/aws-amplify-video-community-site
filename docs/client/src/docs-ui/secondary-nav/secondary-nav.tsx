import {Component, Host, h, Prop, State, Listen} from "@stencil/core";
import {
  secondaryNavStyle,
  hostStyle,
  linkActiveStyle,
  shadowStyle,
} from "./secondary-nav.style";
import {createVNodeFromHyperscriptNode} from "../../utils/hyperscript";
import {pageContext} from "../page/page.context";
import {SelectedFilters} from "../page/page.types";
import {
  AWS_USER_GUIDE,
  IOS_REFERENCE,
  ANDROID_REFERENCE,
  JS_REFERENCE,
} from "../../constants/links";

@Component({tag: "docs-secondary-nav", shadow: false})
export class DocsSecondaryNav {
  /*** the current filter state */
  @Prop() readonly selectedFilters?: SelectedFilters;
  /*** whether or not the current page has a menu */
  @Prop() readonly pageHasMenu?: boolean;

  render() {
    return (
      <Host class={hostStyle} id="secondary-nav">
        <docs-container>
          <div class={secondaryNavStyle}>
            <div>
              <div>
                {[
                  {
                    label: "Getting Started",
                    url: "/start",
                  },
                  {
                    label: "Libraries",
                    url: "/lib",
                    additionalActiveChildRoots: ["/lib", "/sdk"],
                  },
                  {
                    label: "UI Components",
                    url: "/ui",
                    additionalActiveChildRoots: ["/ui"],
                  },
                  {
                    label: "CLI",
                    url: "/cli",
                  },
                  {
                    label: "Console",
                    url: "/console",
                  },
                  {
                    label: "Guides",
                    url: "/guides",
                  },
                  ...(this.selectedFilters?.platform
                    ? [
                        {
                          label: "API Reference",
                          url: (() => {
                            switch (this.selectedFilters.platform) {
                              case "ios": {
                                return IOS_REFERENCE;
                              }
                              case "android": {
                                return ANDROID_REFERENCE;
                              }
                              case "js": {
                                return JS_REFERENCE;
                              }
                            }
                          })(),
                          external: true,
                        },
                      ]
                    : []),
                ].map(({url, label, external, additionalActiveChildRoots}) =>
                  createVNodeFromHyperscriptNode([
                    external ? "amplify-external-link" : "docs-internal-link",
                    {
                      key: label,
                      href: url,
                      ...(external
                        ? {graphic: "black"}
                        : {
                            childActiveClass: linkActiveStyle,
                            additionalActiveChildRoots,
                          }),
                    },
                    ["span", null, label],
                  ]),
                )}
                <div class={shadowStyle}></div>
              </div>
            </div>
          </div>
          {this.pageHasMenu && <amplify-sidebar-open-button />}
        </docs-container>
      </Host>
    );
  }
}

pageContext.injectProps(DocsSecondaryNav, ["selectedFilters"]);
