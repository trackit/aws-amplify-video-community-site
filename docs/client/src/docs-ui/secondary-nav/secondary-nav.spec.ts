import {DocsSecondaryNav} from "./secondary-nav";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-secondary-nav", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsSecondaryNav],
            html: `<docs-secondary-nav />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
