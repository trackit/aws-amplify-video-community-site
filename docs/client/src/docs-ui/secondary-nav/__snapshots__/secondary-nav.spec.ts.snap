// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-secondary-nav Render logic should render 1`] = `
<docs-secondary-nav class="css-ay160a" id="secondary-nav">
  <docs-container>
    <div class="css-16891c2">
      <div>
        <div>
          <docs-internal-link childactiveclass="css-pnqqcs" href="/start">
            <span>
              Getting Started
            </span>
          </docs-internal-link>
          <docs-internal-link childactiveclass="css-pnqqcs" href="/lib">
            <span>
              Libraries
            </span>
          </docs-internal-link>
          <docs-internal-link childactiveclass="css-pnqqcs" href="/ui">
            <span>
              UI Components
            </span>
          </docs-internal-link>
          <docs-internal-link childactiveclass="css-pnqqcs" href="/cli">
            <span>
              CLI
            </span>
          </docs-internal-link>
          <docs-internal-link childactiveclass="css-pnqqcs" href="/console">
            <span>
              Console
            </span>
          </docs-internal-link>
          <docs-internal-link childactiveclass="css-pnqqcs" href="/guides">
            <span>
              Guides
            </span>
          </docs-internal-link>
          <div class="css-1r4lk6l"></div>
        </div>
      </div>
    </div>
  </docs-container>
</docs-secondary-nav>
`;
